keps/sig-scheduling/2458-node-resource-score-strategy/README.md

```diff
@@ -0,0 +1,674 @@
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+The default scheduler includes three score plugins that implement different strategies 
+for preferred resource allocation. Those plugins should not be enabled together, only one.
+
+This KEP proposes to deprecate those plugins and combine them under one Score plugin, the same
+one used for filtering (namely `NodeResourcesFit`), and add a `ScoringStrategy` parameter
+to `NodeResourcesFit` plugin config that allows users to select which exact scoring strategy to run.
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+The motivation is two fold:
+
+1) Reduce the complexity of configuring the default scheduler:
+
+Configuring the scheduler plugins is a tedious task. The relatively large number of plugins
+that the default scheduler supports doesn't make that task easier: more plugins means larger
+number of combinations of enabled/disabled plugins.
+
+Moreover, some combinations don't make sense and potentially harmful. For example, specific
+to this KEP, enabling both least and most allocated scoring plugins is not useful.
+
+Finally, we are planning to add more resource fit scoring strategies, like Best/WrostFit which
+allows preferring nodes with the least/most amount of available absolute resources that can host the
+pod; also, Least/MostAllocatable which allows preferring nodes with the least/most amount of 
+allocatable resources that can host the pod. Adding those strategies as separate plugins will
+make the scheduler configuration problem even worse.
```

Comment:
Is this plugin flexible to support _internal_ combination of sub-fit strategy? Like a user may want to combine 2 strategies: least (absolute) available resource plus most allocated resource (ratio).

Reply:
For new ones, we need to understand if they should be baked into existing scoring strategies, or add knobs to enable them on demand if they are cross cutting (or have keep them in separate plugin).